stumpy/snippets.py

```diff
@@ -110,11 +110,11 @@ def _get_all_profiles(
             S_i,
             T,
             s,
-            M_T[start : start + s],
-            Σ_T[start : start + s],
+            M_T[start : stop - s + 1],
```

Comment:
Why did we need to change this? How does this compare to the original Matlab code?

Reply:
> Why did we need to change this? 

`M_T` should contain the sliding mean of `S_i` (the first argument of this function, i.e. `_mpdist_vect `). Note that `S_i` is set to `S_i = T[start:stop]`. Also, note that `M_T` is the sliding mean of subsequences (with **length `s`**) of `T` (see the line before the for-loop). 

Therefore, the sliding mean of `S_i` will be `M_T[start : stop - s + 1]`. The same is true for `Σ_T ` and `T_subseq_isconstant`. Now, you may wonder why this issue was not exposed before.  The reason is that the slice `start : start + s` covers the slice `start : stop - s + 1` when `s > m/2`. Note that I added new value for s in the test functions. The new values reflect `s < m / 2` scenario and that is why we got error in the first commit. In the second commit, there should have been no error according to my local testing.  I guess the current error in the last push is related to precision. I will investigate it.

> How does this compare to the original Matlab code?

I will check the MATLAB code.

Reply:
> I will check the MATLAB code.

The link provided in the paper is broken, and not working.